src/main/scala/cognite/spark/v1/DataModelInstancesRelation.scala

```diff
@@ -197,81 +194,79 @@ class DataModelInstanceRelation(config: RelationConfig, modelExternalId: String)
           (index, field.name, propertyType)
       }
 
-    def parseRow(row: Row): DataModelInstance = {
+    def parseRow(row: Row): DataModelInstanceCreate = {
       val _ = row.get(externalIdIndex) match {
         case x: String => x
         case _ => throw SparkSchemaHelperRuntime.badRowError(row, "externalId", "String", "")
       }
-      val propertyValues: Map[String, Json] = properties.map {
-        case (index, name, propT) =>
-          name -> tryGetValue(propT._1, propT._2).applyOrElse(
-            row.get(index),
-            (_: Any) =>
-              throw SparkSchemaHelperRuntime
-                .badRowError(row, name, propT._1, "")
-          )
-      }.toMap
-      DataModelInstance(modelExternalId, properties = Some(propertyValues))
+      val propertyValues: Map[String, PropertyType] = properties
+        .map {
+          case (index, name, propT) =>
+            name -> (row.get(index) match {
+              case null if !propT._2 => // scalastyle:off null
+                throw new CdfSparkException(propertyNotNullableMessage(propT._1))
+              case null => // scalastyle:off null
+                None
+              case _ =>
+                Some(
+                  toPropertyType(propT._1).applyOrElse(
+                    row.get(index),
+                    (_: Any) =>
+                      throw SparkSchemaHelperRuntime
+                        .badRowError(row, name, propT._1, "")
+                  ))
+            })
+        }
+        .collect { case (a, Some(value)) => a -> value }
+        .toMap
+
+      DataModelInstanceCreate(modelExternalId, properties = Some(propertyValues))
     }
     parseRow
   }
 }
 
 object DataModelInstanceRelation {
-  private def propertyNotNullableMessage(propertyType: String) =
-    s"Property of $propertyType type is not nullable."
   private def unknownPropertyTypeMessage(a: Any) = s"Unknown property type $a."
 
   private def notValidPropertyTypeMessage(a: Any, propertyType: String) =
```

Comment:
This seems to be thrown when someone tries to write something that can't be parsed to the property type of the schema. We accept multiple types of numbers casted to Float32 for example, perhaps there's a more informative way to let the user know what they need to do!